crates/project_panel/src/project_panel.rs

```diff
@@ -6325,6 +6350,205 @@ mod tests {
         );
     }
 
+    #[gpui::test]
+    async fn test_next_selection_after_delete(cx: &mut gpui::TestAppContext) {
+        init_test_with_editor(cx);
+
+        let fs = FakeFs::new(cx.executor().clone());
+        fs.insert_tree(
+            "/src",
+            json!({
+                "test": {
+                    "first.rs": "// First",
+                    "second.rs": "// Second",
+                    "third.rs": "// Third",
+                }
+            }),
+        )
+        .await;
+
+        let project = Project::test(fs.clone(), ["/src".as_ref()], cx).await;
+        let workspace = cx.add_window(|cx| Workspace::test_new(project.clone(), cx));
+        let cx = &mut VisualTestContext::from_window(*workspace, cx);
+        let panel = workspace.update(cx, ProjectPanel::new).unwrap();
+
+        // Delete middle file (second.rs) - should select next file (third.rs)
+        toggle_expand_dir(&panel, "src/test", cx);
+        select_path(&panel, "src/test/second.rs", cx);
+        submit_deletion(&panel, cx);
```

Comment:
Let's, before every change selection (initial state like here, selection changed after some deletion assert, etc.), assert the way how the tree looked with the selection.
So, there's always "selection before" vs. "selection after" comparison, which makes these tests more readable.

For the tests below too.